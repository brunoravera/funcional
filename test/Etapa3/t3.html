<html>
	<head>
		<link rel="stylesheet" href="gift.css">
	</head>
	<body>
<div class="giftComentario">
  
</div>
<span class="giftNombre">
Sobrecarga_P18_1</span>
<div class="giftFragmento">
Dada la siguiente definicion:</div>

<span class="giftCode">
foo a b c d e = (snd . fst) ((a == e, b < e), c == d)</span>

<div class="giftFragmento">
El tipo mas general es</div>


<div class="giftMO">
\item <div class="giftFragmento">
</div>

<span class="giftCode">
foo :: (Ord a, Eq b) \=> a -> a -> b -> b -> a -> Bool</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
</div>

<span class="giftCode">
foo :: (Ord a, Ord b, Eq c, Eq d, Ord e) \=> a -> b -> c -> d -> e -> Bool</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
</div>

<span class="giftCode">
foo :: (Ord a) \=> a -> a -> a -> a -> a -> Bool</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
No tiene  # Analice el tipo de</div>

<span class="giftCode">
((a \=\= e, b < e), c \=\= d)</span>

<div class="giftFragmento">
. Cuando puedo comparar dos valores?</div>


</div>

<div class="giftFragmento">
</div>


<div class="giftComentario">
 
</div>

<div class="giftComentario">
 
</div>
<span class="giftNombre">
Otras_P18_3</span>
<div class="giftFragmento">
Dada la siguiente definicion:</div>

<span class="giftCode">
map' f = (map (uncurry ($))) . box</span>

<div class="giftFragmento">
¿Cual de las siguientes implementaciones de</div>

<span class="giftCode">
box</span>

<div class="giftFragmento">
hace que</div>

<span class="giftCode">
map'</span>

<div class="giftFragmento">
sea equivalente a</div>

<span class="giftCode">
map</span>

<div class="giftFragmento">
?</div>


<div class="giftShort">
\item <div class="giftFragmento">
</div>

<span class="giftCode">
zip (repeat f)</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
No hay implementacion posible de</div>

<span class="giftCode">
box</span>

<div class="giftFragmento">
para eso.</div>


\item <div class="giftFragmento">
</div>

<span class="giftCode">
zipWith f</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
</div>

<span class="giftCode">
zip f</span>

<div class="giftFragmento">
</div>


</div>

<div class="giftFragmento">
</div>


<div class="giftComentario">
 
</div>

<div class="giftComentario">
 
</div>
<span class="giftNombre">
Otras_P18_4</span>
<div class="giftFragmento">
Dada la siguiente definicion:</div>

<span class="giftCode">
bar f n xs = concat $ map (f n) xs</span>

<div class="giftFragmento">
Cual de las siguientes afirmaciones no es correcta?</div>


<div class="giftMO">
\item <div class="giftFragmento">
</div>

<span class="giftCode">
sum . bar const 1</span>

<div class="giftFragmento">
</div>

<span class="giftMATH">
\equiv</span>

<div class="giftFragmento">
</div>

<span class="giftCode">
length</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
El tipo es</div>

<span class="giftCode">
bar :: (a -> b -> [c]) -> a -> [b] -> [c]</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
</div>

<span class="giftCode">
bar take 2 [[1,2], [3,4,5]]</span>

<div class="giftFragmento">
retorna</div>

<span class="giftCode">
[1,2,3,4]</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
</div>

<span class="giftCode">
bar (\\f x -> [f x])</span>

<div class="giftFragmento">
</div>

<span class="giftMATH">
\equiv</span>

<div class="giftFragmento">
</div>

<span class="giftCode">
map</span>

<div class="giftFragmento">
</div>


</div>

<div class="giftFragmento">
</div>


<div class="giftComentario">
 
</div>

<div class="giftComentario">
 
</div>
<span class="giftNombre">
Algebraicos_P18_5</span>
<div class="giftFragmento">
Cual de las siguientes afirmaciones es correcta?</div>


<div class="giftMO">
\item <div class="giftFragmento">
Dado</div>

<span class="giftCode">
t</span>

<div class="giftFragmento">
de tipo</div>

<span class="giftCode">
A</span>

<div class="giftFragmento">
,</div>

<span class="giftCode">
snd (copia t)</span>

<div class="giftFragmento">
siempre diverge o retorna un entero no negativo</div>


\item <div class="giftFragmento">
El codigo no compila porque la definicion de</div>

<span class="giftCode">
A</span>

<div class="giftFragmento">
es incorrecta</div>


\item <div class="giftFragmento">
El codigo no compila porque la definicion de</div>

<span class="giftCode">
A</span>

<div class="giftFragmento">
es incorrecta</div>


\item <div class="giftFragmento">
El codigo no compila porque la definicion de</div>

<span class="giftCode">
copia</span>

<div class="giftFragmento">
es incorrecta</div>


\item <div class="giftFragmento">
Dado</div>

<span class="giftCode">
t</span>

<div class="giftFragmento">
de tipo</div>

<span class="giftCode">
A</span>

<div class="giftFragmento">
,</div>

<span class="giftCode">
fst (copia t)</span>

<div class="giftFragmento">
siempre diverge o retorna</div>

<span class="giftCode">
t</span>

<div class="giftFragmento">
</div>


</div>

<div class="giftFragmento">
</div>


<div class="giftComentario">
 
 
</div>

<div class="giftComentario">
 
</div>
<span class="giftNombre">
Algebraicos_P18_5</span>
<div class="giftFragmento">
Cual de las siguientes afirmaciones es correcta?</div>


<div class="giftMO">
\item <div class="giftFragmento">
Dado</div>

<span class="giftCode">
t</span>

<div class="giftFragmento">
de tipo</div>

<span class="giftCode">
A</span>

<div class="giftFragmento">
,</div>

<span class="giftCode">
snd (copia t)</span>

<div class="giftFragmento">
siempre diverge o retorna un entero no negativo</div>


\item <div class="giftFragmento">
El codigo no compila porque la definicion de</div>

<span class="giftCode">
A</span>

<div class="giftFragmento">
es incorrecta</div>


\item <div class="giftFragmento">
El codigo no compila porque la definicion de</div>

<span class="giftCode">
copia</span>

<div class="giftFragmento">
es incorrecta</div>


\item <div class="giftFragmento">
Dado</div>

<span class="giftCode">
t</span>

<div class="giftFragmento">
de tipo</div>

<span class="giftCode">
A</span>

<div class="giftFragmento">
,</div>

<span class="giftCode">
fst (copia t)</span>

<div class="giftFragmento">
siempre diverge o retorna</div>

<span class="giftCode">
t</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
El codigo no compila porque la definicion de</div>

<span class="giftCode">
A</span>

<div class="giftFragmento">
es incorrecta</div>


</div>

<div class="giftFragmento">
</div>


<div class="giftComentario">
 
</div>

<div class="giftComentario">
 
</div>
<span class="giftNombre">
Algebraicos_P18_6</span>
<div class="giftFragmento">
Dadas las siguientes definiciones:</div>

<span class="giftCode">
data T = C1 | C2 Int Int | C3 T T</span>

<div class="giftFragmento">
,</div>

<span class="giftCode">
baz C1       = 0</span>

<div class="giftFragmento">
,</div>

<span class="giftCode">
baz (C2 x y) = x</span>

<div class="giftFragmento">
,</div>

<span class="giftCode">
baz (C3 x y) = baz x</span>

<div class="giftFragmento">
.
¿Cual de las siguientes afirmaciones es correcta?</div>


<div class="giftMO">
\item <div class="giftFragmento">
El resultado de evaluar</div>

<span class="giftCode">
baz (C3 (C2 3 4) (C3 (C2 1 2) C1))</span>

<div class="giftFragmento">
es</div>

<span class="giftCode">
3</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
El resultado   de evaluar</div>

<span class="giftCode">
baz (C3 (C2 3 4) (C3 (C2 1 2) C1))</span>

<div class="giftFragmento">
es</div>

<span class="giftCode">
7</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
El   resultado de   evaluar</div>

<span class="giftCode">
baz (C3 (C2 3 4) (C3 (C2 1 2) C1))</span>

<div class="giftFragmento">
es</div>

<span class="giftCode">
7</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
El resultado de evaluar</div>

<span class="giftCode">
baz (C3 (C2 3 4) (C3 (C2 1 2) C1))</span>

<div class="giftFragmento">
es</div>

<span class="giftCode">
1</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
El codigo no compila</div>


</div>

<div class="giftFragmento">
</div>


<div class="giftComentario">
 
</div>

<div class="giftComentario">
 
</div>
<span class="giftNombre">
Fold_P18_7</span>
<div class="giftFragmento">
Dada la siguiente definicion:</div>

<span class="giftCode">
qux f []     ys = ys</span>

<div class="giftFragmento">
</div>

<span class="giftCode">
qux f (x:xs) ys = qux f xs (f x : ys)</span>

<div class="giftFragmento">
Cual de las siguientes afirmaciones es correcta?</div>


<div class="giftMO">
\item <div class="giftFragmento">
</div>

<span class="giftCode">
qux f</span>

<div class="giftFragmento">
</div>

<span class="giftMATH">
\equiv</span>

<div class="giftFragmento">
</div>

<span class="giftCode">
flip $ foldl (\\r x -> f x : r)</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
</div>

<span class="giftCode">
qux f</span>

<div class="giftFragmento">
</div>

<span class="giftMATH">
\equiv</span>

<div class="giftFragmento">
</div>

<span class="giftCode">
foldl (\\r x -> f x : r)</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
</div>

<span class="giftCode">
qux f</span>

<div class="giftFragmento">
</div>

<span class="giftMATH">
\equiv + 2</span>

<div class="giftFragmento">
</div>

<span class="giftCode">
foldl (\\r x -> f x : r)</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
</div>

<span class="giftCode">
qux f</span>

<div class="giftFragmento">
</div>

<span class="giftMATH">
\equiv +  2</span>

<div class="giftFragmento">
</div>

<span class="giftCode">
flip $ foldr (\\x r -> f x : r)</span>

<div class="giftFragmento">
</div>


\item <div class="giftFragmento">
</div>

<span class="giftCode">
qux f</span>

<div class="giftFragmento">
</div>

<span class="giftMATH">
\equiv</span>

<div class="giftFragmento">
</div>

<span class="giftCode">
foldr (\\x r -> f x : r)</span>

<div class="giftFragmento">
</div>


</div>

<div class="giftFragmento">
</div>


<div class="giftComentario">
 
</div>

	</body>
</html>
