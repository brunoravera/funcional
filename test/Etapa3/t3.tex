\documentclass{gift}
\begin{document}
\begin{giftComentario}
  
\end  {giftComentario}
\giftNombre{Sobrecarga_P18_1}
\begin{giftFragmento}
Dada la siguiente definicion:\end  {giftFragmento}

\verb!foo a b c d e = (snd . fst) ((a == e, b < e), c == d)!
\begin{giftFragmento}
El tipo mas general es\end  {giftFragmento}


\begin{giftMO}
\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!foo :: (Ord a, Eq b) \=> a -> a -> b -> b -> a -> Bool!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!foo :: (Ord a, Ord b, Eq c, Eq d, Ord e) \=> a -> b -> c -> d -> e -> Bool!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!foo :: (Ord a) \=> a -> a -> a -> a -> a -> Bool!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
No tiene  # Analice el tipo de\end  {giftFragmento}

\verb!((a \=\= e, b < e), c \=\= d)!
\begin{giftFragmento}
. Cuando puedo comparar dos valores?\end  {giftFragmento}


\end  {giftMO}

\begin{giftFragmento}
\end  {giftFragmento}


\begin{giftComentario}
 
\end  {giftComentario}

\begin{giftComentario}
 
\end  {giftComentario}
\giftNombre{Otras_P18_3}
\begin{giftFragmento}
Dada la siguiente definicion:\end  {giftFragmento}

\verb!map' f = (map (uncurry ($))) . box!
\begin{giftFragmento}
¿Cual de las siguientes implementaciones de\end  {giftFragmento}

\verb!box!
\begin{giftFragmento}
hace que\end  {giftFragmento}

\verb!map'!
\begin{giftFragmento}
sea equivalente a\end  {giftFragmento}

\verb!map!
\begin{giftFragmento}
?\end  {giftFragmento}


\begin{giftShort}
\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!zip (repeat f)!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
No hay implementacion posible de\end  {giftFragmento}

\verb!box!
\begin{giftFragmento}
para eso.\end  {giftFragmento}


\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!zipWith f!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!zip f!
\begin{giftFragmento}
\end  {giftFragmento}


\end  {giftShort}

\begin{giftFragmento}
\end  {giftFragmento}


\begin{giftComentario}
 
\end  {giftComentario}

\begin{giftComentario}
 
\end  {giftComentario}
\giftNombre{Otras_P18_4}
\begin{giftFragmento}
Dada la siguiente definicion:\end  {giftFragmento}

\verb!bar f n xs = concat $ map (f n) xs!
\begin{giftFragmento}
Cual de las siguientes afirmaciones no es correcta?\end  {giftFragmento}


\begin{giftMO}
\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!sum . bar const 1!
\begin{giftFragmento}
\end  {giftFragmento}

$\equiv$
\begin{giftFragmento}
\end  {giftFragmento}

\verb!length!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
El tipo es\end  {giftFragmento}

\verb!bar :: (a -> b -> [c]) -> a -> [b] -> [c]!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!bar take 2 [[1,2], [3,4,5]]!
\begin{giftFragmento}
retorna\end  {giftFragmento}

\verb![1,2,3,4]!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!bar (\\f x -> [f x])!
\begin{giftFragmento}
\end  {giftFragmento}

$\equiv$
\begin{giftFragmento}
\end  {giftFragmento}

\verb!map!
\begin{giftFragmento}
\end  {giftFragmento}


\end  {giftMO}

\begin{giftFragmento}
\end  {giftFragmento}


\begin{giftComentario}
 
\end  {giftComentario}

\begin{giftComentario}
 
\end  {giftComentario}
\giftNombre{Algebraicos_P18_5}
\begin{giftFragmento}
Cual de las siguientes afirmaciones es correcta?\end  {giftFragmento}


\begin{giftMO}
\item \begin{giftFragmento}
Dado\end  {giftFragmento}

\verb!t!
\begin{giftFragmento}
de tipo\end  {giftFragmento}

\verb!A!
\begin{giftFragmento}
,\end  {giftFragmento}

\verb!snd (copia t)!
\begin{giftFragmento}
siempre diverge o retorna un entero no negativo\end  {giftFragmento}


\item \begin{giftFragmento}
El codigo no compila porque la definicion de\end  {giftFragmento}

\verb!A!
\begin{giftFragmento}
es incorrecta\end  {giftFragmento}


\item \begin{giftFragmento}
El codigo no compila porque la definicion de\end  {giftFragmento}

\verb!A!
\begin{giftFragmento}
es incorrecta\end  {giftFragmento}


\item \begin{giftFragmento}
El codigo no compila porque la definicion de\end  {giftFragmento}

\verb!copia!
\begin{giftFragmento}
es incorrecta\end  {giftFragmento}


\item \begin{giftFragmento}
Dado\end  {giftFragmento}

\verb!t!
\begin{giftFragmento}
de tipo\end  {giftFragmento}

\verb!A!
\begin{giftFragmento}
,\end  {giftFragmento}

\verb!fst (copia t)!
\begin{giftFragmento}
siempre diverge o retorna\end  {giftFragmento}

\verb!t!
\begin{giftFragmento}
\end  {giftFragmento}


\end  {giftMO}

\begin{giftFragmento}
\end  {giftFragmento}


\begin{giftComentario}
 
 
\end  {giftComentario}

\begin{giftComentario}
 
\end  {giftComentario}
\giftNombre{Algebraicos_P18_5}
\begin{giftFragmento}
Cual de las siguientes afirmaciones es correcta?\end  {giftFragmento}


\begin{giftMO}
\item \begin{giftFragmento}
Dado\end  {giftFragmento}

\verb!t!
\begin{giftFragmento}
de tipo\end  {giftFragmento}

\verb!A!
\begin{giftFragmento}
,\end  {giftFragmento}

\verb!snd (copia t)!
\begin{giftFragmento}
siempre diverge o retorna un entero no negativo\end  {giftFragmento}


\item \begin{giftFragmento}
El codigo no compila porque la definicion de\end  {giftFragmento}

\verb!A!
\begin{giftFragmento}
es incorrecta\end  {giftFragmento}


\item \begin{giftFragmento}
El codigo no compila porque la definicion de\end  {giftFragmento}

\verb!copia!
\begin{giftFragmento}
es incorrecta\end  {giftFragmento}


\item \begin{giftFragmento}
Dado\end  {giftFragmento}

\verb!t!
\begin{giftFragmento}
de tipo\end  {giftFragmento}

\verb!A!
\begin{giftFragmento}
,\end  {giftFragmento}

\verb!fst (copia t)!
\begin{giftFragmento}
siempre diverge o retorna\end  {giftFragmento}

\verb!t!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
El codigo no compila porque la definicion de\end  {giftFragmento}

\verb!A!
\begin{giftFragmento}
es incorrecta\end  {giftFragmento}


\end  {giftMO}

\begin{giftFragmento}
\end  {giftFragmento}


\begin{giftComentario}
 
\end  {giftComentario}

\begin{giftComentario}
 
\end  {giftComentario}
\giftNombre{Algebraicos_P18_6}
\begin{giftFragmento}
Dadas las siguientes definiciones:\end  {giftFragmento}

\verb!data T = C1 | C2 Int Int | C3 T T!
\begin{giftFragmento}
,\end  {giftFragmento}

\verb!baz C1       = 0!
\begin{giftFragmento}
,\end  {giftFragmento}

\verb!baz (C2 x y) = x!
\begin{giftFragmento}
,\end  {giftFragmento}

\verb!baz (C3 x y) = baz x!
\begin{giftFragmento}
.
¿Cual de las siguientes afirmaciones es correcta?\end  {giftFragmento}


\begin{giftMO}
\item \begin{giftFragmento}
El resultado de evaluar\end  {giftFragmento}

\verb!baz (C3 (C2 3 4) (C3 (C2 1 2) C1))!
\begin{giftFragmento}
es\end  {giftFragmento}

\verb!3!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
El resultado   de evaluar\end  {giftFragmento}

\verb!baz (C3 (C2 3 4) (C3 (C2 1 2) C1))!
\begin{giftFragmento}
es\end  {giftFragmento}

\verb!7!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
El   resultado de   evaluar\end  {giftFragmento}

\verb!baz (C3 (C2 3 4) (C3 (C2 1 2) C1))!
\begin{giftFragmento}
es\end  {giftFragmento}

\verb!7!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
El resultado de evaluar\end  {giftFragmento}

\verb!baz (C3 (C2 3 4) (C3 (C2 1 2) C1))!
\begin{giftFragmento}
es\end  {giftFragmento}

\verb!1!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
El codigo no compila\end  {giftFragmento}


\end  {giftMO}

\begin{giftFragmento}
\end  {giftFragmento}


\begin{giftComentario}
 
\end  {giftComentario}

\begin{giftComentario}
 
\end  {giftComentario}
\giftNombre{Fold_P18_7}
\begin{giftFragmento}
Dada la siguiente definicion:\end  {giftFragmento}

\verb!qux f []     ys = ys!
\begin{giftFragmento}
\end  {giftFragmento}

\verb!qux f (x:xs) ys = qux f xs (f x : ys)!
\begin{giftFragmento}
Cual de las siguientes afirmaciones es correcta?\end  {giftFragmento}


\begin{giftMO}
\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!qux f!
\begin{giftFragmento}
\end  {giftFragmento}

$\equiv$
\begin{giftFragmento}
\end  {giftFragmento}

\verb!flip $ foldl (\\r x -> f x : r)!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!qux f!
\begin{giftFragmento}
\end  {giftFragmento}

$\equiv$
\begin{giftFragmento}
\end  {giftFragmento}

\verb!foldl (\\r x -> f x : r)!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!qux f!
\begin{giftFragmento}
\end  {giftFragmento}

$\equiv + 2$
\begin{giftFragmento}
\end  {giftFragmento}

\verb!foldl (\\r x -> f x : r)!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!qux f!
\begin{giftFragmento}
\end  {giftFragmento}

$\equiv +  2$
\begin{giftFragmento}
\end  {giftFragmento}

\verb!flip $ foldr (\\x r -> f x : r)!
\begin{giftFragmento}
\end  {giftFragmento}


\item \begin{giftFragmento}
\end  {giftFragmento}

\verb!qux f!
\begin{giftFragmento}
\end  {giftFragmento}

$\equiv$
\begin{giftFragmento}
\end  {giftFragmento}

\verb!foldr (\\x r -> f x : r)!
\begin{giftFragmento}
\end  {giftFragmento}


\end  {giftMO}

\begin{giftFragmento}
\end  {giftFragmento}


\begin{giftComentario}
 
\end  {giftComentario}

\end{document}
