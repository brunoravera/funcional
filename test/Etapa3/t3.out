CUESTIONARIO
============

PREGUNTA Sobrecarga_P18_1
COMENTARIO INICIAL:   
[Q [TXT "Dada la siguiente definicion:",CODE "foo a b c d e = (snd . fst) ((a == e, b < e), c == d)",TXT "El tipo mas general es"],A (MO [OK [TXT "",CODE "foo :: (Ord a, Eq b) \\=> a -> a -> b -> b -> a -> Bool",TXT ""],OK [TXT "",CODE "foo :: (Ord a, Ord b, Eq c, Eq d, Ord e) \\=> a -> b -> c -> d -> e -> Bool",TXT ""],NOK [TXT "",CODE "foo :: (Ord a) \\=> a -> a -> a -> a -> a -> Bool",TXT ""],NOK [TXT "No tiene  # Analice el tipo de",CODE "((a \\=\\= e, b < e), c \\=\\= d)",TXT ". Cuando puedo comparar dos valores?"]]),Q [TXT ""]]
COMENTARIO FINAL:  
------------
PREGUNTA Otras_P18_3
COMENTARIO INICIAL:  
[Q [TXT "Dada la siguiente definicion:",CODE "map' f = (map (uncurry ($))) . box",TXT "\191Cual de las siguientes implementaciones de",CODE "box",TXT "hace que",CODE "map'",TXT "sea equivalente a",CODE "map",TXT "?"],A (SHORT [OK [TXT "",CODE "zip (repeat f)",TXT ""],OK [TXT "No hay implementacion posible de",CODE "box",TXT "para eso."],OK [TXT "",CODE "zipWith f",TXT ""],OK [TXT "",CODE "zip f",TXT ""]]),Q [TXT ""]]
COMENTARIO FINAL:  
------------
PREGUNTA Otras_P18_4
COMENTARIO INICIAL:  
[Q [TXT "Dada la siguiente definicion:",CODE "bar f n xs = concat $ map (f n) xs",TXT "Cual de las siguientes afirmaciones no es correcta?"],A (MO [NOK [TXT "",CODE "sum . bar const 1",TXT "",MATH "\\equiv",TXT "",CODE "length",TXT ""],NOK [TXT "El tipo es",CODE "bar :: (a -> b -> [c]) -> a -> [b] -> [c]",TXT ""],NOK [TXT "",CODE "bar take 2 [[1,2], [3,4,5]]",TXT "retorna",CODE "[1,2,3,4]",TXT ""],NOK [TXT "",CODE "bar (\\\\f x -> [f x])",TXT "",MATH "\\equiv",TXT "",CODE "map",TXT ""]]),Q [TXT ""]]
COMENTARIO FINAL:  
------------
PREGUNTA Algebraicos_P18_5
COMENTARIO INICIAL:  
[Q [TXT "Cual de las siguientes afirmaciones es correcta?"],A (MO [OK [TXT "Dado",CODE "t",TXT "de tipo",CODE "A",TXT ",",CODE "snd (copia t)",TXT "siempre diverge o retorna un entero no negativo"],NOK [TXT "El codigo no compila porque la definicion de",CODE "A",TXT "es incorrecta"],NOK [TXT "El codigo no compila porque la definicion de",CODE "A",TXT "es incorrecta"],NOK [TXT "El codigo no compila porque la definicion de",CODE "copia",TXT "es incorrecta"],NOK [TXT "Dado",CODE "t",TXT "de tipo",CODE "A",TXT ",",CODE "fst (copia t)",TXT "siempre diverge o retorna",CODE "t",TXT ""]]),Q [TXT ""]]
COMENTARIO FINAL:  
 
------------
PREGUNTA Algebraicos_P18_5
COMENTARIO INICIAL:  
[Q [TXT "Cual de las siguientes afirmaciones es correcta?"],A (MO [OK [TXT "Dado",CODE "t",TXT "de tipo",CODE "A",TXT ",",CODE "snd (copia t)",TXT "siempre diverge o retorna un entero no negativo"],NOK [TXT "El codigo no compila porque la definicion de",CODE "A",TXT "es incorrecta"],NOK [TXT "El codigo no compila porque la definicion de",CODE "copia",TXT "es incorrecta"],NOK [TXT "Dado",CODE "t",TXT "de tipo",CODE "A",TXT ",",CODE "fst (copia t)",TXT "siempre diverge o retorna",CODE "t",TXT ""],NOK [TXT "El codigo no compila porque la definicion de",CODE "A",TXT "es incorrecta"]]),Q [TXT ""]]
COMENTARIO FINAL:  
------------
PREGUNTA Algebraicos_P18_6
COMENTARIO INICIAL:  
[Q [TXT "Dadas las siguientes definiciones:",CODE "data T = C1 | C2 Int Int | C3 T T",TXT ",",CODE "baz C1       = 0",TXT ",",CODE "baz (C2 x y) = x",TXT ",",CODE "baz (C3 x y) = baz x",TXT ".\n\191Cual de las siguientes afirmaciones es correcta?"],A (MO [OK [TXT "El resultado de evaluar",CODE "baz (C3 (C2 3 4) (C3 (C2 1 2) C1))",TXT "es",CODE "3",TXT ""],NOK [TXT "El resultado   de evaluar",CODE "baz (C3 (C2 3 4) (C3 (C2 1 2) C1))",TXT "es",CODE "7",TXT ""],NOK [TXT "El   resultado de   evaluar",CODE "baz (C3 (C2 3 4) (C3 (C2 1 2) C1))",TXT "es",CODE "7",TXT ""],NOK [TXT "El resultado de evaluar",CODE "baz (C3 (C2 3 4) (C3 (C2 1 2) C1))",TXT "es",CODE "1",TXT ""],NOK [TXT "El codigo no compila"]]),Q [TXT ""]]
COMENTARIO FINAL:  
------------
PREGUNTA Fold_P18_7
COMENTARIO INICIAL:  
[Q [TXT "Dada la siguiente definicion:",CODE "qux f []     ys = ys",TXT "",CODE "qux f (x:xs) ys = qux f xs (f x : ys)",TXT "Cual de las siguientes afirmaciones es correcta?"],A (MO [OK [TXT "",CODE "qux f",TXT "",MATH "\\equiv",TXT "",CODE "flip $ foldl (\\\\r x -> f x : r)",TXT ""],NOK [TXT "",CODE "qux f",TXT "",MATH "\\equiv",TXT "",CODE "foldl (\\\\r x -> f x : r)",TXT ""],NOK [TXT "",CODE "qux f",TXT "",MATH "\\equiv + 2",TXT "",CODE "foldl (\\\\r x -> f x : r)",TXT ""],NOK [TXT "",CODE "qux f",TXT "",MATH "\\equiv +  2",TXT "",CODE "flip $ foldr (\\\\x r -> f x : r)",TXT ""],NOK [TXT "",CODE "qux f",TXT "",MATH "\\equiv",TXT "",CODE "foldr (\\\\x r -> f x : r)",TXT ""]]),Q [TXT ""]]
COMENTARIO FINAL:  
------------
